firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -111,22 +123,53 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
    * to complete the download and installation.
    */
   @NonNull
-  public synchronized Task<Void> updateToLatestRelease() {
-    if (cachedUpdateToLatestReleaseTask != null && !cachedUpdateToLatestReleaseTask.isComplete()) {
+  public synchronized UpdateTask updateIfNewReleaseAvailable() {
+    synchronized (updateTaskLock) {
+      if (cachedUpdateToLatestReleaseTask != null
+          && !cachedUpdateToLatestReleaseTask.isComplete()) {
+        return cachedUpdateToLatestReleaseTask;
+      }
+
+      cachedUpdateToLatestReleaseTask = new UpdateTaskImpl();
+    }
+    checkForNewRelease()
+        .onSuccessTask(
```

Comment:
Any particular reason we are using an executor here? If we do need an executor, we should create one for the main thread. I don't see why we need to implement this on a separate thread.

Reply:
removed executor